src/main/java/com/scaler/lld/design/assignments/prototype/ConfigurationRegister.java

```diff
@@ -0,0 +1,23 @@
+package com.scaler.lld.design.assignments.prototype;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class ConfigurationRegister implements ConfigurationPrototypeRegistry{
+
+    private Map<ConfigurationType, Configuration> register = new ConcurrentHashMap<>();
+    @Override
+    public void addPrototype(Configuration user) {
+        register.put(user.getType(), user);
+    }
+
+    @Override
+    public Configuration getPrototype(ConfigurationType type) {
+        return register.get(type);
+    }
+
+    @Override
+    public Configuration clone(ConfigurationType type) {
+        return register.get(type).cloneObject();
```

Comment:
An edge you should handle here is the prototype not being present in the registry.
Try to use the `Optional` class instead of an if-else block.

Reply:
Thank you for alerting that but still we should use if-else right, checking ifPresent or not in Optional object? Is this use case for better readability or has some other advantage?

Reply:
In this scenario, it is mostly for readability. Indentation reduces the readability of a piece of code.

In a little more complex cases, you can use the optional interface to trigger further processing and even throw exceptions. [Reference](https://www.baeldung.com/java-optional#or-else-throw)